bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/domain/model/GridFlow.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solarwatt.internal.domain.model;
+
+import static org.openhab.binding.solarwatt.internal.SolarwattBindingConstants.CHANNEL_CURRENT_LIMIT;
+import static org.openhab.binding.solarwatt.internal.SolarwattBindingConstants.CHANNEL_FEED_IN_LIMIT;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.solarwatt.internal.domain.SolarwattChannel;
+import org.openhab.binding.solarwatt.internal.domain.dto.DeviceDTO;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+
+/**
+ * Class planning the energy flow out/into the grid.
+ *
+ * This fields have been identified to exist:
+ * com.kiwigrid.kiwiapp.gridflow.GridFlow=[
+ * ConfigTimeshifting,
+ * ConfigEvStationControl,
+ * FractionPVTestLimit,
+ * ConfigInverterControl,
+ * LogLevel,
+ * PowerSetpoint,
+ * ConfigPeakshaving,
+ * ToEMRequestTag,
+ * CurrentLimit,
+ * ConfigBatteryControl,
+ * ConfigForecastBatteryControl,
+ * ConfigEvStationChargingControl,
+ * ConfigSgReady,
+ * ToCloudDataTag
+ * ]
+ *
+ * @author Sven Carstens - Initial contribution
+ */
+@NonNullByDefault
+public class GridFlow extends Device {
+    public static final String solarWattClassname = "com.kiwigrid.kiwiapp.gridflow.GridFlow";
+    private final Logger logger = LoggerFactory.getLogger(GridFlow.class);
+
+    private @Nullable ConfigInverterControl configInverterControl;
+
+    public GridFlow(DeviceDTO deviceDTO) {
+        super(deviceDTO);
+    }
+
+    @Override
+    public void update(DeviceDTO deviceDTO) {
+        super.update(deviceDTO);
+
+        this.addAmpereQuantity(CHANNEL_CURRENT_LIMIT, deviceDTO, true);
+
+        try {
+            JsonObject rawConfigInverterControl = deviceDTO.getJsonObjectFromTag("ConfigInverterControl");
+            Gson gson = new GsonBuilder().create();
+            this.configInverterControl = gson.fromJson(rawConfigInverterControl, GridFlow.ConfigInverterControl.class);
+        } catch (Exception ex) {
+            this.configInverterControl = null;
+            this.logger.error("Could not read ConfigInverterControl", ex);
```

Comment:
error is for critical errors see

https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Reduced loglevels.